File: app/src/main/java/com/pgshare/studentroomsharingapp/MainActivity.java
package com.pgshare.studentroomsharingapp;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.google.firebase.auth.FirebaseAuth;
import com.pgshare.studentroomsharingapp.Adapter.Owner;
import com.pgshare.studentroomsharingapp.Authentication.Login;
import com.pgshare.studentroomsharingapp.Authentication.OwnerLogin;

public class MainActivity extends AppCompatActivity {

    private FirebaseAuth firebaseAuth;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        getSupportActionBar().setBackgroundDrawable(new ColorDrawable(getResources().getColor(R.color.C_color)));
        // Initialize Firebase Authentication
        firebaseAuth = FirebaseAuth.getInstance();
        Log.d("temp_debug", "Testing log filter");
    }

    public void Owner(View view) {
        Intent intent = new Intent(MainActivity.this, OwnerLogin.class);
        startActivity(intent);
    }

    public void Rent(View view) {
        Intent intent = new Intent(MainActivity.this, Login.class);
        startActivity(intent);
    }

    // Inflate the menu resource file
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.main_menu, menu);
        getSupportActionBar().setBackgroundDrawable(new ColorDrawable(getResources().getColor(R.color.C_color)));
        return true;
    }

    // Handle menu item selection
    @SuppressLint("NonConstantResourceId")
    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        int id = item.getItemId();
        if (id == R.id.profile) {
            if (firebaseAuth.getCurrentUser() != null) {
                openProfile();
            } else {
                Toast.makeText(this, "Please log in to view your profile", Toast.LENGTH_SHORT).show();
            }
            return true;
        } else if (id == R.id.contact) {
            // Handle contact menu item click
            openContactUs();
            return true;
        } else if (id == R.id.menu_About) {
            // Handle about menu item click
            openAbout();
            return true;
        } else if (id == R.id.menu_logout) {
            // Handle logout menu item click
            logoutUser();
            return true;
        } else {
            return super.onOptionsItemSelected(item);
        }
    }


    private void openAbout() {
        /*Intent intent = new Intent(MainActivity.this, About.class);
        startActivity(intent);*/
        Toast.makeText(this, "Working on it", Toast.LENGTH_SHORT).show();
    }

    private void openContactUs() {
       /* Intent intent = new Intent(MainActivity.this, ContactUs.class);
        startActivity(intent);*/
        Toast.makeText(this, "Working on it", Toast.LENGTH_SHORT).show();
    }

 /*   private void openContactUs() {
        Intent intent = new Intent(MainActivity.this, ContactUs.class);
        startActivity(intent);
    }*/

    private void openProfile() {
        Intent intent = new Intent(MainActivity.this, Profile_Page.class);
        Toast.makeText(this, "Opening Profile", Toast.LENGTH_SHORT).show();
        startActivity(intent);
    }

    // Method to log out the current user
    private void logoutUser() {
        // Sign out the user
        firebaseAuth.signOut();

        // Redirect to the login activity
        Toast.makeText(this, "Logging out", Toast.LENGTH_SHORT).show();
        // Clear the back stack to prevent the user from navigating back to the main activity after logout
    }
}

File: app/src/main/java/com/pgshare/studentroomsharingapp/MainActivity.kt
package com.pgshare.studentroomsharingapp

import android.content.Intent
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.pgshare.studentroomsharingapp.BuildConfig
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import com.google.firebase.auth.FirebaseAuth
import com.pgshare.studentroomsharingapp.Authentication.Login
import com.pgshare.studentroomsharingapp.repository.FirebaseRepository
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity() {
    private var firebaseAuth: FirebaseAuth? = null
    private val repository = FirebaseRepository()


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        firebaseAuth = FirebaseAuth.getInstance()

        findViewById<android.widget.TextView>(R.id.tvContinueAsGuest)?.setOnClickListener {
            startActivity(Intent(this, StudentDashboardActivity::class.java))
        }
    }

    fun Rent(view: View?) {
        val intent = Intent(this@MainActivity, StudentDashboardActivity::class.java)
        startActivity(intent)
    }

    fun Owner(view: View?) {
        val user = firebaseAuth?.currentUser
        if (user != null) {
            lifecycleScope.launch {
                val role = repository.getUserRole(user.uid)
                if (role == "owner") {
                    val intent = Intent(this@MainActivity, AddRoomActivity::class.java)
                    startActivity(intent)
                } else {
                    Toast.makeText(this@MainActivity, "Only property owners can list rooms", Toast.LENGTH_SHORT).show()
                }
            }
        } else {
            MaterialAlertDialogBuilder(this)
                .setTitle("Login Required")
                .setMessage("You need to log in to list a room. Would you like to log in?")
                .setPositiveButton("Log In") { _, _ ->
                    startActivity(Intent(this@MainActivity, Login::class.java))
                }
                .setNegativeButton("Cancel", null)
                .show()
        }
    }

    // Inflate the menu resource file
    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.main_menu, menu)
        supportActionBar?.setBackgroundDrawable(ColorDrawable(resources.getColor(R.color.colorPrimary)))
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId
        if (id == R.id.action_profile) {
            if (firebaseAuth?.currentUser != null) {
                openProfile()
            } else {
                Toast.makeText(this, "Please log in to view your profile", Toast.LENGTH_SHORT)
                    .show()
            }
            return true
        } else if (id == R.id.action_contact_us) {
            openContactUs()
            return true
        } else if (id == R.id.action_about) {
            openAbout()
            return true
        } else if (id == R.id.action_logout) {
            logoutUser()
            return true
        } else {
            return super.onOptionsItemSelected(item)
        }
    }


    private fun openAbout() {
        MaterialAlertDialogBuilder(this)
            .setTitle(R.string.about)
            .setMessage("${getString(R.string.app_name)} v${BuildConfig.VERSION_NAME}\n\n${getString(R.string.copyright)}")
            .setPositiveButton(R.string.close, null)
            .show()
    }

    private fun openContactUs() {
        val intent = Intent(Intent.ACTION_SEND).apply {
            type = "message/rfc822"
            putExtra(Intent.EXTRA_EMAIL, arrayOf("support@studentroomsharing.app"))
            putExtra(Intent.EXTRA_SUBJECT, "Support Request - ${getString(R.string.app_name)}")
        }
        if (intent.resolveActivity(packageManager) != null) {
            startActivity(intent)
        } else {
            Toast.makeText(this, R.string.no_email_app, Toast.LENGTH_SHORT).show()
        }
    }

    private fun openProfile() {
        val intent = Intent(this, StudentDashboardActivity::class.java).apply {
            putExtra("TARGET_FRAGMENT", "PROFILE")
        }
        startActivity(intent)
    }

    private fun logoutUser() {
        firebaseAuth?.signOut()
        startActivity(Intent(this, Login::class.java).apply {
            flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
        })
        finish()
    }
}
